### PR TITLE
Fixes supermatter not consuming simple mobs inside simple mob holders

### PR DESCRIPTION
**Pull request by jknpj**
Also stop people from power from holders.

:cl: 
 * bugfix: Fixes a bug with supermatter not consuming simple mobs hold by humans

### CODE REVIEW

**Comment by PJB3005** [comment]
I disagree with the way this is implemented.

Deletion of the holder without context should delete the contained mob no questions asked. All code like dropping holders should just take care to eject the mob beforehand so that doesn't happen.

**Comment by AI-Is-Rogue** [comment]
>being hold by humans

 👌

File: code/modules/supermatter/supermatter.dm [hunk 1]
```diff
@@ -390,12 +386,16 @@
 
 	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, 1)
 
+	if(istype(W, /obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = W
+		if(H.stored_mob)
+			Consume(H.stored_mob)
+
```
Comment (N3X15) [inline code/modules/supermatter/supermatter.dm:393]: This whole thing is a bit too snowflakey for me.  The holder is the one that has the odd behaviour, and it should be the one to perform whatever steps are needed to correct any weird shit that happens when it's deleted.  For example, moving this logic to Destroy().

File: code/modules/supermatter/supermatter.dm [hunk 2]
```diff
@@ -415,16 +415,22 @@
 
 	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, 1)
 
+	if(istype(AM, /obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = AM
+		if(H.stored_mob)
+			Consume(H.stored_mob)
```
Comment (N3X15) [inline code/modules/supermatter/supermatter.dm:421]: Same here.

File: code/modules/supermatter/supermatter.dm [hunk 3]
```diff
@@ -415,16 +415,22 @@
...
 	if(istype(user))
 		user.dust()
-		if(istype(user,/mob/living/simple_animal/mouse)) //>implying mice are going to follow the rules
+		if(ismouse(user)) //>implying mice are going to follow the rules
 			return
 		power += 200
 	else
+		if(istype(user, /obj/item/weapon/holder)) //STOP MAKING POWER OUT OF HOLDERS HOLY SHIT
+			qdel(user)
+			return
```
Comment (N3X15) [inline code/modules/supermatter/supermatter.dm:433]: Immediately after this block, it calls qdel anyway.  There should be a istypeinlist() check afterward that which returns on match, or sets the power modifier from 200 to 0, or whatever tickles your fancy.

**Review by N3X15 — CHANGES_REQUESTED** [review]

**Comment by jknpj** [comment]
I'll see if i can do something about this messing with holders, not with supermatter

**Pull request closed without merging**